lib/fwup/stream.ex

```diff
@@ -52,27 +52,27 @@ defmodule Fwup.Stream do
     {:noreply, state}
   end
 
-  def handle_info({_port, {:data, <<"ER", code::16, message::binary>>}}, state) do
-    dispatch(state, {:error, code, message})
-    {:noreply, state}
-  end
-
   def handle_info({_port, {:data, <<"WN", code::16, message::binary>>}}, state) do
     dispatch(state, {:warning, code, message})
     {:noreply, state}
   end
 
-  def handle_info({_port, {:data, <<"OK", code::16, message::binary>>}}, state) do
-    dispatch(state, {:ok, code, message})
-    {:noreply, state}
+  def handle_info({port, {:data, <<"ER", code::16, message::binary>>}}, state) do
+    dispatch(state, {:error, code, message})
+    _ = Port.command(port, "ER")
+    _ = Port.close(port)
+    {:stop, message, %{state | port: nil}}
   end
 
-  def handle_info({_port, {:exit_status, 0}}, state) do
+  def handle_info({port, {:data, <<"OK", code::16, message::binary>>}}, state) do
+    dispatch(state, {:ok, code, message})
+    _ = Port.command(port, "OK")
+    _ = Port.close(port)
     {:stop, :normal, %{state | port: nil}}
   end
 
   def handle_info({_port, {:exit_status, status}}, state) do
-    {:stop, {:exit, status}, %{state | port: nil}}
+    {:stop, {:error, {:unexpected_exit, status}}, %{state | port: nil}}
```

Comment:
Won't we still get the exit_status here even if we are manually doing `Port.close` and it exits normal?

Reply:
the clause that does the `Port.close` returns `{:stop, :normal, %{state | port: nil}` so i don't think the exit status message will get handled